src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -2874,6 +3044,365 @@ static void F(bool c, IOut<object> x, IOut<object?> y)
+interface IOut<out T> { }
+class C
+{
+    public static I<object> operator+(I<object> x, C y) => x;
+    public static IIn<object> operator+(IIn<object> x, C y) => x;
+    public static IOut<object> operator+(IOut<object> x, C y) => x;
+    static void F(C c, I<object> x, I<object?> y)
+    {
+        x += c;
+        y += c;
+    }
+    static void F(C c, IIn<object> x, IIn<object?> y)
+    {
+        x += c;
+        y += c;
+    }
+    static void F(C c, IOut<object> x, IOut<object?> y)
+    {
+        x += c;
+        y += c;
+    }
+}";
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (12,9): warning CS8619: Nullability of reference types in value of type 'I<object?>' doesn't match target type 'I<object>'.
+                //         y += c;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y").WithArguments("I<object?>", "I<object>").WithLocation(12, 9),
+                // (12,9): warning CS8619: Nullability of reference types in value of type 'I<object>' doesn't match target type 'I<object?>'.
+                //         y += c;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y += c").WithArguments("I<object>", "I<object?>").WithLocation(12, 9),
+                // (17,9): warning CS8619: Nullability of reference types in value of type 'IIn<object>' doesn't match target type 'IIn<object?>'.
+                //         y += c;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y += c").WithArguments("IIn<object>", "IIn<object?>").WithLocation(17, 9),
+                // (22,9): warning CS8619: Nullability of reference types in value of type 'IOut<object?>' doesn't match target type 'IOut<object>'.
+                //         y += c;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y").WithArguments("IOut<object?>", "IOut<object>").WithLocation(22, 9));
+        }
+
+        // PROTOTYPE(NullableReferenceTypes): Assign each of the deconstructed values.
+        [Fact(Skip = "TODO")]
+        public void IdentityConversion_DeconstructionAssignment()
+        {
+            var source =
+@"interface IIn<in T> { }
+interface IOut<out T> { }
+class C<T>
+{
+    void Deconstruct(out IIn<T> x, out IOut<T> y)
+    {
+        throw new System.NotImplementedException();
+    }
+    static void F(C<object> c)
+    {
+        IIn<object?> x;
+        IOut<object?> y;
+        (x, y) = c;
+    }
+    static void G(C<object?> c)
+    {
+        IIn<object> x;
+        IOut<object> y;
+        (x, y) = c;
+    }
+}";
+            var comp = CreateCompilation(
+                source,
+                parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (13,18): warning CS8619: Nullability of reference types in value of type 'C<object>' doesn't match target type '(IIn<object?> x, IOut<object?> y)'.
```

Comment:
These diagnostics are confusing.

Reply:
Added `// PROTOTYPE` comment.

---
In reply to: [190951330](https://github.com/dotnet/roslyn/pull/26356#discussion_r190951330) [](ancestors = 190951330)